### PR TITLE
feat: preparation for onboarding tasks to artifact hub

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
One of the requirement for onboarding tasks to artifact hub is, to have a special folder with released manifests, which is then followed by artifacthub.
This PR adds a new step in release pipeline, which copies generated manifests from tasks/ into released-tasks folder with new tag. Then this change will be added as a part of PR to main branch (artifacthub is following main by default), which updates repo with new release tag. 
The result tree of released-tasks folder will look like this: 
```
release/tasks/cleanup-vm/
├── 0.17.0
│   ├── cleanup-vm.yaml
│   ├── examples
│   │   ├── secrets
│   │   │   ├── ssh-secret-advanced.yaml
│   │   │   └── ssh-secret.yaml
│   │   └── taskruns
│   │       ├── cleanup-vm-simple-taskrun.yaml
│   │       └── cleanup-vm-with-ssh-taskrun.yaml
│   └── README.md
└── 0.18.0
    ├── cleanup-vm.yaml
    ├── examples
    │   ├── secrets
    │   │   ├── ssh-secret-advanced.yaml
    │   │   └── ssh-secret.yaml
    │   └── taskruns
    │       ├── cleanup-vm-simple-taskrun.yaml
    │       └── cleanup-vm-with-ssh-taskrun.yaml
    └── README.md
```
It also refactors labels and annotations in each task to match requirements of artifact hub

**Release note**:
```
copy released manifests into released-tasks folder
```

### CODE REVIEW

File: scripts/copy-released-manifests.sh
Comment (codingben) [inline scripts/copy-released-manifests.sh:12]: IMO change `released-tasks` to `release`. ArtifactHub URL is already long so let's make it shorter.

File: scripts/copy-released-manifests.sh
Comment (codingben) [inline scripts/copy-released-manifests.sh:13]: I'm not sure if it should be `vX.Y.Z` or `X.Y.Z`. I'd recommend to keep it as `X.Y.Z` if possible, because this is also what other projects do, e.g. buildah [1].

[1] https://github.com/openshift-pipelines/tektoncd-catalog/tree/p/tasks/task-buildah

**Review by codingben — CHANGES_REQUESTED** [review]

**Comment by codingben** [comment]
Can you run ArtifactHub linter before it's copying them? I think it would be required to do so, it helps to catch issues earlier than going through PR cycles again.

[1] https://artifacthub.io/docs/topics/cli (`ah lint ...`)

File: artifacthub-repo.yml
Comment (codingben) [inline artifacthub-repo.yml:1]: Please consider setting value here or remove this file. I don't think it's needed right now.

**Comment by codingben** [comment]
It looks good, nice! just a few comments and I'll do final review before adding `/lgtm`.

File: scripts/copy-released-manifests.sh
Comment (0xFelix) [reply scripts/copy-released-manifests.sh:17]: Can we move them entirely to this repo instead? If the way forward is to deploy all Tekton resources from ArtifactHub instead from SSP?

Comment (ksimon1) [reply scripts/copy-released-manifests.sh:17]: These comments are work which has to be done after the merge. The repo currently does not contain pipelines which will be here in the future. After we will copy them here, these lines can be uncommented, and pipelines will be part of each release

File: scripts/copy-released-manifests.sh
Comment (0xFelix) [inline scripts/copy-released-manifests.sh:20]: `-` for default value in variable expansion missing?

File: scripts/copy-released-manifests.sh
Comment (0xFelix) [inline scripts/copy-released-manifests.sh:12]: Same. Did you run shellcheck against this?

File: scripts/run-ah-linter.sh
Comment (0xFelix) [inline scripts/run-ah-linter.sh:7]: Use `jq` solution without piping into sort, tail and tr?

**Comment by ksimon1** [comment]
@0xFelix the pipelines will be moved to this repo. It will be follow up work after this PR

File: scripts/run-ah-linter.sh
Comment (ksimon1) [reply scripts/run-ah-linter.sh]: fixed

File: scripts/copy-released-manifests.sh
Comment (ksimon1) [reply scripts/copy-released-manifests.sh:20]: this means cut first character of the string (in this case cut 'v' from 'vx.y.z')

**Comment by 0xFelix** [comment]
/lgtm
/hold

Giving @codingben a chance to review again

**Comment by codingben** [comment]
Looks good, nice work! :) Welcome to ArtifactHub 😄 

/lgtm

**Pull request merged**
